juju/juju
# Relate via

## Description of change

We now support adding a remote relation and specifying the egress subnets from which outbound traffic will originate. This compliments the existing model wide config attribute "egress-subnets".

The first commit renames the model attribute "egress-cidrs" to "egress-subnets".
The second commit addes the --via support.

The UX is "juju add-relation foo bar --via cidr1,cidr2,..."

## QA steps

juju bootstrap aws dbctrl
juju deploy mysql
juju offer mysql:db

From behind a firewall or cgnat:
juju bootstrap lxd mwctrl
juju deploy mediawiki
juju relate mediawiki:db dbctrl:defauly.mysql --via <what's-my-ip>/32

check aws firewall ports and check that media wiki is operational
